lab_1/main.py

```diff
@@ -86,10 +183,43 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
+    # Check for bad input
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profile_to_compare, dict)
+            or not isinstance(top_n, int)):
+        return None
+    # Find common words, while preserving the order
+    unknown_top = get_top_n_words(unknown_profile["freq"], top_n)
+    compare_top = get_top_n_words(profile_to_compare["freq"], top_n)
+    common = []
+    for token in compare_top:
+        if token in unknown_top:
+            common.append(token)
+    # Sort common words alphabetically
+    sorted_common = sorted(common)
+    # Find score
+    score = len(common) / len(compare_top)
+    # Token length
+    tokens = [token for token, freq in profile_to_compare["freq"].items()]
+    tokens = sorted(tokens, key=len)
+    min_length_word = tokens[0]
+    max_length_word = tokens[-1]
+    average_token_length = sum(map(len, tokens)) / len(tokens)
```

Comment:
not bad, can you explain in writing what happens here?

Reply:
This question is outdated, as this part of the code has been rewritten (to reduce computational complexity as per he big O notation), but it has not been marked as such.